scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -265,7 +268,9 @@ private static Optional<ResultsProtocol.State> getRequestState(
               request.reason()
                   .orElseThrow(() -> new Error("Unexpected state: %s request state has no failure message".formatted(request.status()))),
               request.analysisId());
-          case SUCCESS -> new ResultsProtocol.State.Success(getResults(connection, request.analysisId()), request.analysisId());
+          case SUCCESS -> new ResultsProtocol.State.Success(getResults(connection, request.analysisId()),
+                                                            request.analysisId(),
+                                                            request.datasetId().orElseThrow(() -> new Error("Unexpected state: %s request state has no dataset id".formatted(request.status()))));
```

Comment:
From what I understand from https://github.com/NASA-AMMOS/aerie/pull/708 : let's say that simulation was disabled for all goals, there will be no activity in the simulation. Code from this PR takes activities from the simulation and tries to match them up with activities from the scheduler's plan. As simulation was disabled, it will not insert any new spans. But it will insert simulation results that are inconsistent with the state of the plan as it will compute the simulation results on the basis of the initial plan. I could add a step where we verify that there is a 1:1 correspondance between activities in the schedule and in the simulation before we store these results.

Reply:
If there is no correspondance, I am not sure what dataset id we can return.

Reply:
_If_ the user understands that the dataset posted to the DB may be inaccurate to the state of the plan if they turn off sim for some goals, I don't think we need to do 1:1 correspondence check. So this should be fine unless the Scheduler ever uses the last Simulation Results as its base, in which case `datasetId` should probably be `null`, since Scheduling didn't produce a dataset.

Reply:
Yeah it is gonna be null unless there is a 1:1 correspondence.